src/_pytest/python_api.py

```diff
@@ -527,8 +540,35 @@ def _is_numpy_array(obj):
 
 # builtin pytest.raises helper
 
+_E = TypeVar("_E", bound=BaseException)
+
+
+@overload
+def raises(
+    expected_exception: Union["Type[_E]", Tuple["Type[_E]", ...]],
+    *,
+    match: Optional[Union[str, Pattern]] = ...  # noqa: W504 (SyntaxError in Python 3.5)
```

Comment:
changing `...` to `None` fixes this, definitely a bug in `pycodestyle` but if we can work around it let's :+1:

Reply:
actually, we should be using `extend-ignore` instead of `ignore` in `tox.ini` for `flake8` -- can probably fix that instead

Reply:
here's a fix from `pycodestyle`'s side: https://github.com/PyCQA/pycodestyle/pull/875

Reply:
So would you like to change to `None` or keep the noqa and remove it once pycodestyle is updated?

Reply:
`s/ignore/extend-ignore/` should be sufficient to make this go away, then you can keep the code

Reply:
Ah I understand, because `W504` is ignored by default, then `extend-ignore` will keep it ignored.

I'll add a commit for that.